pywinauto/unittests/test_keyboard.py

```diff
@@ -1,179 +1,214 @@
+# License along with this library; if not, write to the
+#    Free Software Foundation, Inc.,
+#    59 Temple Place,
+#    Suite 330,
+#    Boston, MA 02111-1307 USA
+
+"""Module containing tests for SendKeys Module"""
+
 from __future__ import unicode_literals
 from __future__ import print_function
 
 import sys
 import os
-#import locale
 import unittest
-#import six
+import subprocess
+import time
 sys.path.append(".")
-
-from pywinauto.keyboard import SendKeys, KeySequenceError
-from pywinauto.keyboard import KeyAction, VirtualKeyAction, PauseAction
-from pywinauto.sysinfo import is_x64_Python, is_x64_OS
-from pywinauto.application import Application
-#from pywinauto.actionlogger import ActionLogger
-from pywinauto.timings import Timings
-
-
-mfc_samples_folder = os.path.join(
-   os.path.dirname(__file__), r"..\..\apps\MFC_samples")
-if is_x64_Python():
-    mfc_samples_folder = os.path.join(mfc_samples_folder, 'x64')
+if sys.platform == 'win32':
+    from pywinauto.keyboard import SendKeys, KeySequenceError
+    from pywinauto.keyboard import KeyAction, VirtualKeyAction, PauseAction
+    from pywinauto.sysinfo import is_x64_Python, is_x64_OS
+    from pywinauto.application import Application
+    from pywinauto import backend
+else:
+    parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    sys.path.insert(0, parent_dir)
+    import mouse
+    send_keys_dir = os.path.join(parent_dir, r"linux/")
+    sys.path.insert(0, send_keys_dir)
+    from pywinauto.keyboard import SendKeys, KeySequenceError, KeyAction
+    import clipboard
+
+def mfc_samples():
+    mfc_samples_folder = os.path.join(
+       os.path.dirname(__file__), r"..\..\apps\MFC_samples")
+    if is_x64_Python():
+        mfc_samples_folder = os.path.join(mfc_samples_folder, 'x64')
+    return mfc_samples_folder
 
 def _notepad_exe():
     if is_x64_Python() or not is_x64_OS():
         return r"C:\Windows\System32\notepad.exe"
     else:
         return r"C:\Windows\SysWOW64\notepad.exe"
 
+def _test_app():
+    test_folder = os.path.join(os.path.dirname
+                               (os.path.dirname
+                                (os.path.dirname
+                                 (os.path.abspath(__file__)))),
+                               r"apps/SendKeysTester")
+    return os.path.join(test_folder, r"send_keys_test_app")
 
 class SendKeysTests(unittest.TestCase):
-
-    """Unit tests for the Sendkeys module"""
+    "Unit tests for the Sendkeys module"
 
     def setUp(self):
-        """Set some data and ensure the application is in the state we want"""
-        Timings.Defaults()
-        self.app = Application()
-        self.app.start(_notepad_exe())
-
-        self.dlg = self.app.UntitledNotepad
-        self.ctrl = self.dlg.Edit
+        """Start the application set some data and ensure the application
+        is in the state we want it."""
+        if sys.platform == 'win32':
+            backend.activate("native")

```

Comment:
This line fails the tests. Please remove as not needed. As well as `from pywinauto import backend`.